actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -840,6 +840,7 @@ def url_for(options, route_name = nil, url_strategy = UNKNOWN, method_name = nil
 
       def call(env)
         req = make_request(env)
+        req.parameters
```

Comment:
Calling `parameters` earlier should probably not impact performance as the results get calculated once and are memoized in the header.

Reply:
Yeah, that was my thinking as well.